scaffolding-chef/lib/scaffolding.ps1

```diff
@@ -0,0 +1,133 @@
+  Chef-Client
+}
+"@
+}
+
+
+function Invoke-DefaultBuild {
+  if(-Not (Test-Path -Path "${scaffold_policyfile_path}")){
+    Write-BuildLine "Could not detect a policyfiles directory, this is required to proceed!"
+    exit 1
+  }
+
+  Remove-Item "${scaffold_policyfile_path}/*.lock.json" -Force
+  $policyfile="${scaffold_policyfile_path}/${scaffold_policy_name}.rb"
+
+  Get-Content $policyfile | ? { $_.StartsWith("include_policy")} | % {
+    $p = $_.Split()[1]
+    $p = $p.Replace("`"", "").Replace(",", "")
+    Write-BuildLine "Detected included policyfile, ${p}.rb, installing"
+    chef install "${scaffold_policyfile_path}/${p}.rb"
+  }
+  Write-BuildLine "Installing ${policyfile}"
+  chef install "${policyfile}"
+}
+
+function Invoke-DefaultInstall {
+  Write-BuildLine "Exporting Chef Infra Repository"
+  chef export "${scaffold_policyfile_path}/${scaffold_policy_name}.lock.json" "${pkg_prefix}"
+
+  Write-BuildLine "Creating Chef Infra configuration"
+  New-Item -ItemType directory -Path "${pkg_prefix}/config"
+  Add-Content -Path "$pkg_prefix/.chef/config.rb" -Value @"
+cache_path "$($ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath("$pkg_svc_data_path/cache").Replace("\","/"))"
+node_path "$($ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath("$pkg_svc_data_path/nodes").Replace("\","/"))"
+role_path "$($ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath("$pkg_svc_data_path/roles").Replace("\","/"))"
+chef_zero.enabled true
+ENV['PSModulePath'] += "C:/Program\ Files/WindowsPowerShell/Modules"
+"@
+
+  Write-BuildLine "Creating initial bootstrap configuration"
+  Copy-Item -Path "$pkg_prefix/.chef/config.rb" -Destination "$pkg_prefix/config/bootstrap-config.rb"
+  Add-Content -Path "$pkg_prefix/config/bootstrap-config.rb" -Value @"
+ENV['PATH'] += ";C:/WINDOWS;C:/WINDOWS/system32/;C:/WINDOWS/system32/WindowsPowerShell/v1.0;C:/ProgramData/chocolatey/bin"
+"@
+
+  Write-BuildLine "Creating Chef Infra client configuration"
+  Copy-Item -Path "$pkg_prefix/.chef/config.rb" -Destination "$pkg_prefix/config/client-config.rb"
+  Add-Content -Path "$pkg_prefix/config/client-config.rb" -Value @"
+ssl_verify_mode {{cfg.ssl_verify_mode}}
+ENV['PATH'] += "{{cfg.env_path_prefix}}"
+
+{{#if cfg.data_collector.enable ~}}
+chef_guid "{{sys.member_id}}"
+data_collector.token "{{cfg.data_collector.token}}"
+data_collector.server_url "{{cfg.data_collector.server_url}}"
+{{/if ~}}
+"@
+
+  Write-BuildLine "Generating config/attributes.json"
+  Add-Content -Path "$pkg_prefix/config/attributes.json" -Value @"
+{{#if cfg.attributes}}
+{{toJson cfg.attributes}}
+{{else ~}}
+{}
+{{/if ~}}
+"@
+
+  Write-BuildLine "Generating Chef Habiat configuration, default.toml"
+  Add-Content -Path "$pkg_prefix/default.toml" -Value @"
+interval = 1800
+splay = 1800
+splay_first_run = 0
+run_lock_timeout = 1800
+log_level = "warn"
+chef_client_ident = "" # this is blank by default so it can be populated from the bind
```

Comment:
Did we ever decide on this? I'm not seeing the config value used in the run hook so I'm not sure this is needed.

Reply:
It still needs work :( I'll take it out for now.